shadowsocks/udp.go

```diff
@@ -120,8 +127,10 @@ func (s *udpService) Start() {
 				}
 				return onet.NewConnectionError("ERR_READ", "Failed to read from client", err)
 			}
-			defer logger.Debugf("UDP done with %v", clientAddr.String())
-			logger.Debugf("UDP Request from %v with %v bytes", clientAddr, clientProxyBytes)
+			if logger.IsEnabledFor(logging.DEBUG) {
+				defer logger.Debugf("UDP(%v): done", clientAddr)
+				debugUDPAddr(clientAddr, "Outbound packet has %d bytes", clientProxyBytes)
```

Comment:
Since this is already in the DEBUG block, you can revert to `logger.Debugf`.

Reply:
Done